driver/terraform.go

```diff
@@ -170,6 +170,14 @@ func (tf *Terraform) InitTask(ctx context.Context) error {
 	return tf.initTask(ctx)
 }
 
+// DeleteTask deletes the task on the driver
+func (tf *Terraform) DeleteTask(ctx context.Context) {
+	tf.mu.Lock()
+	defer tf.mu.Unlock()
+
+	tf.deleteTask(ctx)
+}
```

Comment:
The public method, I think, makes sense as DeleteTask, it could do more in the future theoretically. It is called from the server TaskDelete only. Could also be called DeleteTask cleanup maybe? It's kinda like we're destructing the driver before deleting it from the map

I've renamed the private method to deregisterTemplate() however which I think is a bit clearer, does that make sense?

Reply:
For posterity, we chatted about this async (feel free to correct if I misrepresent anything):
- Agreed that that renaming the private method to `deregisterTemplate()` works
- For the public method, it may do more than deregister the template in the future so the naming shouldn't have parity with the private method
- `DeleteTask` won't quite literally delete the task from the driver - it's more of a destructor for the driver
- Brainstormed names: `Destruct` and `TearDown`

I also just thought of `Cleanup` and `Destroy` - even `DestroyTask` might work 🤔 it kind of pairs well with existing `InitTask`. If this method will support future options to delete the task's directories, files, and resources, then destroy might be fitting. No strong preferences from me at the moment. Let me know what your thoughts are

Reply:
Went with DestroyTask, I agree it pairs well with InitTask